ZenPacks/zenoss/Microsoft/Windows/datasources/ShellDataSource.py

```diff
@@ -814,39 +808,43 @@ def parse_result(config, result):
                         if dp_value is not None:
                             parsed_results['values'][dsconf.component][dp_id] = dp_value, 'N'
             # Maps:
-            ag_om = None
             ag_info_maps = ag_results.get('ag_info')
-            if isinstance(ag_info_maps, dict):
-                ag_om = ObjectMap()
-                ag_om.id = dsconf.params['instanceid']
-                ag_om.title = dsconf.params['contexttitle']
-                ag_om.compname = dsconf.params['contextcompname']
-                ag_om.modname = dsconf.params['contextmodname']
-                ag_om.relname = dsconf.params['contextrelname']
-                sql_instance_data = {
-                    'quorum_state': lookup_ag_quorum_state(
-                        ag_results.get('cl_quorum_state'))
-                }
-                ag_om = fill_ag_om(ag_om, ag_info_maps, prepId, sql_instance_data)
-                parsed_results['maps'].append(ag_om)
+            if not ag_info_maps:
+                log.debug('Got empty maps response for Availability Group {}.'.format(ag_name))
+                # We got empty result for particular Availability Group - seems like it is unreachable.
+                # Need to return set of properties with default values
+                ag_info_maps = get_default_properties_value_for_component('WinSQLAvailabilityGroup')
+
+            ag_om = ObjectMap()
+            ag_om.id = dsconf.params['instanceid']
+            ag_om.title = dsconf.params['contexttitle']
+            ag_om.compname = dsconf.params['contextcompname']
+            ag_om.modname = dsconf.params['contextmodname']
+            ag_om.relname = dsconf.params['contextrelname']
+            sql_instance_data = {
+                'quorum_state': lookup_ag_quorum_state(
+                    ag_results.get('cl_quorum_state'))
+            }
+            ag_om = fill_ag_om(ag_om, ag_info_maps, prepId, sql_instance_data)
+            parsed_results['maps'].append(ag_om)
+
             # Events:
             # 1. check whether Primary replica SQL Instance has changed. If yes - create an event.
-            if ag_om:
```

Comment:
No need this check as we check [this](https://github.com/zenoss/ZenPacks.zenoss.Microsoft.Windows/pull/1150/files#diff-e0d2db946d6c9d028116c3f0691aa319a3046026b95b3f3abd89da1ad7a21d05R834)